staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -43,10 +44,15 @@ import (
 	"k8s.io/kubectl/pkg/util/templates"
 )
 
+const (
+	MsgAnnotationUnchanged = "annotation(s) unchanged" // no change
+	MsgAnnotationModified  = "annotation(s) modified"  // support ADD, DEL and ADD && DEL scenarios
+)
+
 // AnnotateOptions have the data required to perform the annotate operation
 type AnnotateOptions struct {
 	PrintFlags *genericclioptions.PrintFlags
-	PrintObj   printers.ResourcePrinterFunc
+	PrintObj   func(string) printers.ResourcePrinterFunc
```

Comment:
We have an issue due to this usage in `label` https://github.com/kubernetes/kubectl/issues/1265. Could you please check this issue is also happening for `annotate` command with this change?. Not a blocker for me but that would be great if you can add some validation mechanism for that(unit test, integration test, etc.)

Reply:
Tested locally, solved this https://github.com/kubernetes/kubectl/issues/1265 with this PR.
Doc separator '---' works well both for label and annotate.

Reply:
Since this PR is fixing it, can we please include regression tests for both commands?

Reply:
Updated with regression tests for both commands.
PTAL thanks.